src/lib/zf/attr.c

```diff
@@ -300,13 +316,13 @@ int zf_attr_get_str(struct zf_attr* attr, const char* name, char** val)
   switch( f->type ) {
   case zf_attr_type_str: {
     char** p = get_field_str(attr, f);
-    if( *p != NULL ) {
+    if( strcmp(f->name, "log_file")==0 )
+      *val = strdup(zf_log_file_name);
+    else if( *p != NULL ) {
       *val = strdup(*p);
       if( *val == NULL )
         return -ENOMEM;
     }
-    else
-      *val = NULL;
```

Comment:
Do we want to be deleting this else statement? I think it still makes sense in the case of some string field not being set. I expect a nice app would probably do something like:
```c
char* val = NULL;
int rc = zf_attr_get_str(..., &val)
if (rc < 0) return rc;
if (val) {
  // do something
}
```
But I worry specifically about the case where the consumer does not set `val = NULL`, especially as we were kind enough to do this for them previously.

Reply:
agreed.